lib/common/schemas.c

```diff
@@ -1236,38 +1250,20 @@ pcmk_update_configured_schema(xmlNode **xml, bool to_logs)
                 || (schema->schema_index < orig_version)) {
                 // We couldn't validate any schema at all
-                if (to_logs) {
-                    pcmk__config_err("Cannot upgrade configuration (claiming "
-                                     "%s schema) to at least %s because it "
-                                     "does not validate with any schema from "
-                                     "%s to the latest",
-                                     pcmk__s(original_schema_name, "no"),
-                                     x_0_schema->name, effective_original_name);
-                } else {
-                    fprintf(stderr, "Cannot upgrade configuration (claiming "
-                                    "%s schema) to at least %s because it "
-                                    "does not validate with any schema from "
-                                    "%s to the latest\n",
-                                    pcmk__s(original_schema_name, "no"),
-                                    x_0_schema->name, effective_original_name);
-                }
+                out->err(out, "Cannot upgrade configuration (claiming "
+                              "%s schema) to at least %s because it "
+                              "does not validate with any schema from "
+                              "%s to the latest",
+                              pcmk__s(original_schema_name, "no"),
+                              x_0_schema->name, effective_original_name);
             } else {
                 // We updated configuration successfully, but still too low
-                if (to_logs) {
-                    pcmk__config_err("Cannot upgrade configuration (claiming "
-                                     "%s schema) to at least %s because it "
-                                     "would not upgrade past %s",
-                                     pcmk__s(original_schema_name, "no"),
-                                     x_0_schema->name,
-                                     pcmk__s(new_schema_name, "unspecified version"));
-                } else {
-                    fprintf(stderr, "Cannot upgrade configuration (claiming "
-                                    "%s schema) to at least %s because it "
-                                    "would not upgrade past %s\n",
-                                    pcmk__s(original_schema_name, "no"),
-                                    x_0_schema->name,
-                                    pcmk__s(new_schema_name, "unspecified version"));
-                }
+                out->err(out, "Cannot upgrade configuration (claiming "
+                              "%s schema) to at least %s because it "
+                              "would not upgrade past %s",
+                              pcmk__s(original_schema_name, "no"),
+                              x_0_schema->name,
+                              pcmk__s(new_schema_name, "unspecified version"));
```

Comment:
We don't actually need to pass an output object to this function, because `pcmk__config_warn()` and `pcmk_config_err()` already use the handlers set by `pcmk__set_config_error_handler()` and `pcmk__set_config_warning_handler()`, and crm_verify sets output-object-based handlers. All we should have to do is use `pcmk__config_warn()` and `pcmk__config_err()` unconditionally (for crm_verify, which sets the handlers, the messages will go to the output object, while for any other callers, they will go to the logs).